java/build/src/main/java/sleeper/build/maven/dependencydraw/DrawDependencyGraph.java

```diff
@@ -38,45 +46,123 @@
 public class DrawDependencyGraph {
     public boolean showTransitiveDependencies = false;
 
+    LayoutPaintable.BalloonRings balloonLayoutRings;
+    LayoutPaintable.RadialRings radialLayoutRings;
+
+    String instructions =
+            "<html>"
+                    + "<h3>Graph Transformation:</h3>"
+                    + "<ul>"
+                    + "<li>Mousewheel scales with a crossover value of 1.0.<p>"
+                    + "     - scales the graph layout when the combined scale is greater than 1<p>"
+                    + "     - scales the graph view when the combined scale is less than 1"
+                    + "<li>Mouse1+drag pans the graph"
+                    + "<li>Mouse1 double click on the background resets all transforms"
+                    + "</ul>"
+                    + "<h3>Vertex/Edge Selection:</h3>"
+                    + "<ul>"
+                    + "<li>Mouse1+MENU on a vertex or edge selects the vertex or edge and deselects any others"
+                    + "<li>Mouse1+MENU+Shift on a vertex toggles selection of the vertex"
+                    + "<li>Mouse1+MENU on a selected edge toggles selection of the edge"
+                    + "<li>Mouse1+MENU+drag elsewhere selects vertices in a region"
+                    + "<li>Mouse1+Shift+drag adds selection of vertices in a new region"
+                    + "</ul>"
+                    + "<h3>Vertex Transformation:</h3>"
+                    + "<ul>"
+                    + "<li>Mouse1+MENU+drag on a selected vertex moves all selected Vertices"
+                    + "</ul>"
+                    + "Note that MENU == Command on a Mac, MENU == CTRL on a PC"
+                    + "</html>";
+
     public void drawGraph(GraphModel model) {
-        DirectedGraph<GraphNode, GraphEdge> g = new DirectedSparseGraph<>();
+        Graph<GraphNode, GraphEdge> g =
+                GraphTypeBuilder.<GraphNode, GraphEdge>directed().buildGraph();
         model.getNodes().forEach(g::addVertex);
-        model.getEdges().forEach(edge -> g.addEdge(edge, edge.getFrom(model), edge.getTo(model)));
-        KKLayout<GraphNode, GraphEdge> layout = new KKLayout<>(g);
-        layout.setLengthFactor(1.2);
+        model.getEdges().forEach(edge -> g.addEdge(edge.getFrom(model), edge.getTo(model), edge));
+
+        Dimension size = new Dimension(600, 600);
+        VisualizationViewer<GraphNode, GraphEdge> vv =
+                VisualizationViewer.builder(g)
+                        .viewSize(size)
+                        .layoutSize(size).build();
```

Comment:
SpotBugs is flagging up a problem here. VisualizationViewer.builder doesn't populate the type parameter for the type of builder it returns (B), so the compiler doesn't know the type of the object that's returned from viewSize or layoutSize.

Do you think that could be sorted in the library? You can suppress the warning with `@SuppressFBWarnings("BC_UNCONFIRMED_CAST_OF_RETURN_VALUE")`.